tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/circuitbreaker/clientserver/CircuitBreakerClientRollingWindow.java

```diff
@@ -0,0 +1,89 @@
+/*
+ *******************************************************************************
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.eclipse.microprofile.fault.tolerance.tck.circuitbreaker.clientserver;
+
+import java.io.Serializable;
+import java.sql.Connection;
+
+import javax.enterprise.context.RequestScoped;
+
+import org.eclipse.microprofile.faulttolerance.CircuitBreaker;
+/**
+ * A client to exercise Circuit Breaker thresholds, with a SuccessThreshold of 2,
+ * a requestVolumeThreshold of 4, failureRatio of 0.5 and a 1 millisecond delay
+ *
+ * @author <a href="mailto:emijiang@uk.ibm.com">Emily Jiang</a>
+ */
+@RequestScoped
+public class CircuitBreakerClientRollingWindow implements Serializable {
+        
+    private static final long serialVersionUID = 1L;
+    private int counterForInvokingServiceA = 0;
```

Comment:
Issue #323 requested that we don't reuse circuit breakers between tests. (We were expecting each test to have it's own new request scope and for that to mean we got a new circuit breaker instance).

I wonder if that means we should avoid relying on each test having a fresh request scope? If so we should have one counter for each test method here. (or we should reset the counter to zero at the start of each test, but that seems more prone to error)

Reply:
ah. My c+p mistake. I should have used two different counters. fixed now.

Reply:
That private counterForInvokingServiceA attribute is counting calls to the same bean instance. 
Different requests scoped should see different CircuitBreakerClientDefaultSuccessThreshold objects.
In order to be effective that attribute would need to be static.
If we are seeing a non static attribute being incremented, it means its being called from the same context.
This makes me think the all the requests are being made from the same scope making the test ineffective. 
What do you guys think?
In order to be

Reply:
@brunobat I don't think I'm quite following you. You're correct in that the counter will only count calls to the same instance, but the test is making all of its calls against a single instance, so I'd expect it to work. Within one test method, I'd only expect there to be one request context, so we'd only get one instance of the request scoped bean.

The original code assumed that each test method will have a different request context as it relied on each test using a different instance to ensure that the counter was zero at the start of the test. This is true in most cases, but some embedded containers will run all tests in the same request context.

In the current code, `counterForInvokingService1` counts invocations of `service1RollingWindowOpenAfter4()`. `testRollingWindowCircuitOpen()` makes five requests to this method against the same instance of `CircuitBreakerClientRollingWindow`, and then checks the counter.

Reply:
Ok, I was mixing `@RequestScoped` with `@ApplicationScoped` In this case the test uses `@RequestScope` as we have been doing.